Common/DeviceSchema/DeviceSchemaHelperND.cs

```diff
@@ -0,0 +1,304 @@
+        }
+
+        /// <summary>
+        /// _rid is used internally by the DocDB and is required for use with DocDB.
+        /// (_rid is resource id)
+        /// </summary>
+        /// <param name="device">Device data</param>
+        /// <returns>_rid property value as string, or empty string if not found</returns>
+        public static string GetDocDbRid(DeviceND device)
+        {
+            return SchemaHelperND.GetDocDbRid<DeviceND>(device);
+        }
+
+        /// <summary>
+        /// id is used internally by the DocDB and is sometimes required.
+        /// </summary>
+        /// <param name="device">Device data</param>
+        /// <returns>Value of the id, or empty string if not found</returns>
+        public static string GetDocDbId(DeviceND device)
+        {
+            return SchemaHelperND.GetDocDbId<DeviceND>(device);
+        }
+
+        /// <summary>
+        /// Build a valid device representation in the dynamic format used throughout the app.
+        /// </summary>
+        /// <param name="deviceId"></param>
+        /// <param name="isSimulated"></param>
+        /// <param name="iccid"></param>
+        /// <returns></returns>
+        public static DeviceND BuildDeviceStructure(string deviceId, bool isSimulated, string iccid)
+        {
+            DeviceND device = new DeviceND();
+
+            InitializeDeviceProperties(device, deviceId, isSimulated);
+            InitializeSystemProperties(device, iccid);
+
+            device.Commands = new List<Command>();
+            device.CommandHistory = new List<CommandHistoryND>();
+            device.IsSimulatedDevice = isSimulated;
+
+            return device;
+        }
+
+        /// <summary>
+        /// Initialize the device properties for a new device.
+        /// </summary>
+        /// <param name="device"></param>
+        /// <param name="deviceId"></param>
+        /// <param name="isSimulated"></param>
+        /// <returns></returns>
+        public static void InitializeDeviceProperties(DeviceND device, string deviceId, bool isSimulated)
+        {
+            DeviceProperties deviceProps = new DeviceProperties();
+            deviceProps.DeviceID = deviceId;
+            deviceProps.HubEnabledState = null;
+            deviceProps.CreatedTime = DateTime.UtcNow;
+            deviceProps.DeviceState = "normal";
+            deviceProps.UpdatedTime = null;
+
+            device.DeviceProperties = deviceProps;
+        }
+
+        /// <summary>
+        /// Initialize the system properties for a new device.
+        /// </summary>
+        /// <param name="device"></param>
+        /// <param name="iccid"></param>
+        /// <returns></returns>
+        public static void InitializeSystemProperties(DeviceND device, string iccid)
+        {
+            SystemProperties systemProps = new SystemProperties();
+            systemProps.ICCID = iccid;
+
+            device.SystemProperties = systemProps;
```

Comment:
that is object assignment, you don't need an overloader, if that's what you meant